python/pyiceberg/expressions/literals.py

```diff
@@ -346,6 +365,12 @@ class TimeLiteral(Literal[int]):
     def __init__(self, value: int):
         super().__init__(value, int)
 
+    def increment(self) -> TimeLiteral:
+        return TimeLiteral(self.value + 1)
```

Comment:
I don't think that we allow time literals in partition expressions, do we? That seems weird because time literals are bounded between 0 and at the number of microseconds in a day. I'd probably exclude this.

Reply:
Makes sense, removed 👍🏻